lib/TPP/ConvertCheckToLoops.cpp

```diff
@@ -0,0 +1,115 @@
+//===- ConvertTppToLoops.cpp ------------------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "TPP/Dialect/Check/CheckOps.h"
+#include "TPP/Passes.h"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "mlir/Dialect/ControlFlow/IR/ControlFlowOps.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Math/IR/Math.h"
+#include "mlir/Dialect/MemRef/IR/MemRef.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+using namespace mlir;
+using namespace mlir::check;
+using namespace mlir::cf;
+using namespace mlir::arith;
+using namespace mlir::math;
+
+#define GEN_PASS_CLASSES
+#include "TPP/Passes.h.inc"
+#include <iostream>
+namespace {
+
+struct ConvertAlmostEqualsOp
+    : public OpRewritePattern<check::ExpectAlmostEqOp> {
+  using OpRewritePattern<check::ExpectAlmostEqOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(check::ExpectAlmostEqOp almostEqOp,
+                                PatternRewriter &rewriter) const override {
+    Location loc = almostEqOp.getLoc();
+    SmallVector<Value> ubs;
+    size_t rank = almostEqOp.getLhs().getType().cast<MemRefType>().getRank();
+    for (size_t idx = 0; idx < rank; idx++) {
+      Value dim = rewriter.create<arith::ConstantIndexOp>(
+          loc,
+          almostEqOp.getLhs().getType().cast<MemRefType>().getShape()[idx]);
```

Comment:
I don't recall but we all the check op have static shape right? If we allow dynamic you may want to have something like: https://github.com/plaidml/tpp-sandbox/commit/3e5f2db8b1e51dbb85d7474f4e53e97caa611dda